app/v2/createUpdatePeriodsOfAccount/CreateUpdatePeriodsOfAccountService.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.createUpdatePeriodsOfAccount
+
+import api.controllers.RequestContext
+import api.models.errors.{
+  BusinessIdFormatError,
+  InternalError,
+  MtdError,
+  NinoFormatError,
+  NotFoundError,
+  RuleCessationDateError,
+  RuleEndBeforeStartDateError,
+  RuleEndDateError,
+  RuleIncorrectGovTestScenarioError,
+  RuleOutsideAmendmentWindowError,
+  RulePeriodsOverlapError,
+  RuleStartDateError,
+  RuleTaxYearNotEndedError,
+  RuleTaxYearNotSupportedError,
+  TaxYearFormatError
+}
+import api.services.{BaseService, ServiceOutcome}
+import cats.implicits.toBifunctorOps
+import v2.createUpdatePeriodsOfAccount.request.CreateUpdatePeriodsOfAccountRequestData
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class CreateUpdatePeriodsOfAccountService @Inject() (connector: CreateUpdatePeriodsOfAccountConnector) extends BaseService {
+
+  def create(request: CreateUpdatePeriodsOfAccountRequestData)(implicit ctx: RequestContext, ec: ExecutionContext): Future[ServiceOutcome[Unit]] =
+    connector.create(request).map(_.leftMap(mapDownstreamErrors(downstreamErrorMap)))
+
+  private val downstreamErrorMap: Map[String, MtdError] =
```

Comment:
5009 is still missing

Reply:
5009 should not be mapped in the service as it will never get there. 401 is not [here](https://github.com/hmrc/business-details-api/blob/main/app/api/connectors/httpparsers/StandardDownstreamHttpParser.scala#L64). Also we are not adding in the STP as it is always gonna be mapped to a 500